File: Project Euler/10. Summation of primes/summation_of_primes.cc
#include <iostream>
#include <cmath>

using namespace std;

int sum_of_primes(int num){
    int sum = 0;
    int half = int(sqrt(num)) +1;
    
    for (int i =2;i < num + 1; i++){
        for (int j=2; j < half; j++){
            if (i % j == 0){
                break;
            }
        cout << i << " ";
        sum += i;
        }
    }
    return sum;
}


int main(){
    int num = 10;
    cout << "\n10001st prime number is " << sum_of_primes(num) << "." << endl;
    return 0;
}

File: Project Euler/10. Summation of primes/summation_of_primes.py
""" The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
"""

from math import sqrt

def sum_of_primes(num):
    for x in range(2, num+1):
        half = int(sqrt(x)) + 1
        for y in range(2, half):
            if x % y == 0:
                break
        else:
            yield x


def main(num):
    return sum([x for x in sum_of_primes(num)])

# REVIEW: Need to be optimized.